js/filter.js

```diff
@@ -2422,12 +2449,34 @@ class SourceFilter extends Filter {
 		if (reprintedFilter) reprintedFilter.setValue("Reprinted", 0);
 	}
 
+	_setSort(sortType = "DEFAULT") {
+		this._isItemsDirty = true;
+		this._itemSortFn = this._SORT_OPTIONS[sortType];
+		this._meta.sort = sortType;
+	}
+
 	static getCompleteFilterSources (ent) {
 		return ent.otherSources
 			? [ent.source].concat(ent.otherSources.map(src => new SourceFilterItem({item: src.source, isIgnoreRed: true, isOtherSource: true})))
 			: ent.source;
 	}
 
+	_doRenderPills () {
+		const sort = this._meta.sort;
+		if (sort) { 
+			this._setSort(sort);
+		}
+		super._doRenderPills();
```

Comment:
a bit long-winded, try:

```suggestion
		if (this._meta.sort) this._setSort(sort);
		super._doRenderPills();
```